src/plugins/kibana_usage_collection/server/collectors/saved_objects_counts/kibana_usage_collector.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Observable } from 'rxjs';
+import type { ElasticsearchClient, SharedGlobalConfig } from 'src/core/server';
+import type { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { take } from 'rxjs/operators';
+import { snakeCase } from 'lodash';
+import { getSavedObjectsCounts } from './get_saved_object_counts';
+
+interface KibanaSavedObjectCounts {
+  dashboard: { total: number };
+  visualization: { total: number };
+  search: { total: number };
+  index_pattern: { total: number };
+  graph_workspace: { total: number };
+  timelion_sheet: { total: number };
+}
+
+interface KibanaUsage extends KibanaSavedObjectCounts {
+  index: string;
+}
+
+const TYPES = [
+  'dashboard',
+  'visualization',
+  'search',
+  'index-pattern',
+  'graph-workspace',
+  'timelion-sheet',
+];
+
+export async function getKibanaSavedObjectCounts(
+  esClient: ElasticsearchClient,
+  kibanaIndex: string
+): Promise<KibanaSavedObjectCounts> {
+  const buckets = await getSavedObjectsCounts(esClient, kibanaIndex, TYPES);
+
+  const allZeros = (Object.fromEntries(
+    TYPES.map((type) => [snakeCase(type), { total: 0 }])
+  ) as unknown) as KibanaSavedObjectCounts;
+
+  return buckets.reduce((acc, { key, doc_count: total = 0 }) => {
+    const type = snakeCase(key) as keyof KibanaSavedObjectCounts;
+    acc[type].total += total;
+    return acc;
+  }, allZeros);
+}
+
+export function registerKibanaUsageCollector(
+  usageCollection: UsageCollectionSetup,
+  legacyConfig$: Observable<SharedGlobalConfig>
+) {
+  usageCollection.registerCollector(
+    usageCollection.makeUsageCollector<KibanaUsage>({
+      type: 'kibana',
```

Comment:
This is the pre-existing collector. But git thinks it's a new file after changing the name of the folder and moving some logic in here to repurpose `getSavedObjectCounts` to a more generic method.

Reply:
I created the follow-up issue https://github.com/elastic/kibana/issues/99464 to eventually remove it in the future when it's safe.